tests/services/explain/explain_test.py

```diff
@@ -0,0 +1,113 @@
+from google.cloud.forseti.services.explain.service import GrpcExplainerFactory
+from google.cloud.forseti.services.inventory.service import GrpcInventoryFactory
+from google.cloud.forseti.services.model.service import GrpcModellerFactory
+from google.cloud.forseti.services.server import InventoryConfig
+from tests.services.inventory import gcp_api_mocks
+from tests.services.api_tests.api_tester import ModelTestRunner
+from tests.services.api_tests.model_test import MODEL
+from tests.services.util.db import create_test_engine
+from tests.unittest_utils import ForsetiTestCase
+
+
+class TestServiceConfig(object):
+    """ServiceConfig stub."""
+
+    def __init__(self, inventory_config):
+        self.engine = create_test_engine()
+        self.model_manager = ModelManager(self.engine)
+        self.inventory_config = inventory_config
+
+    def run_in_background(self, function):
+        """Stub."""
+        function()
+        return self
+
+    def get_engine(self):
+        """Stub."""
+        return self.engine
+
+
+def create_tester(inventory_config):
+    """Creates a model based test runner.
+    
+    Args:
+        inventory_config (InventoryConfig): the inventory configuration
+
+    Returns:
+        ModelTestRunner: the model test runner
+
+    """
+    return ModelTestRunner(
+        MODEL, TestServiceConfig(inventory_config),
+        [
+            GrpcExplainerFactory,
+            GrpcInventoryFactory,
+            GrpcModellerFactory,
+        ])
+
+
+class ExplainerTest(ForsetiTestCase):
+    """Test based on declarative model."""
+
+    def setUp(self):
+        pass
+
+    def test_explain_is_supported(self):
+        """Test explain is supported."""
+
+        inventory_config_with_organization_root = (
+            InventoryConfig(gcp_api_mocks.ORGANIZATION_ID, '', {}, '')                                                   )
+        self.setup = create_tester(
+            inventory_config_with_organization_root)
+
+        def test(client):
+            """Test implementation with API client."""
+            expected_reply = set([
+                                 'role/a',
+                                 'role/b',
+                                 'role/c',
+                                 'role/d'
+                                 ])
+            actual_reply = client.explain.list_roles('')
+            self.assertEqual(expected_reply,
+                             set(actual_reply.role_names))
+
+        self.setup.run(test)
+
+    def test_explain_is_not_supported(self):
+        """Test explain is not supported."""
+
+        inventory_config_with_folder_root = (
+            InventoryConfig(gcp_api_mocks.FOLDER_ID, '', {}, '')                                                   )
+        self.setup = create_tester(inventory_config_with_folder_root)
+
+        def test(client):
+            """Test implementation with API client."""
+            try:
+                client.explain.list_roles('')
```

Comment:
Nit: you could use self.assertRaisesRegex here instead. https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaisesRegexp.

Reply:
Great tip, didn't know about this before.